scripts/new-upstream-snapshot

```diff
@@ -163,14 +166,14 @@ is_merge_commit() {
 
 main() {
     local short_opts="hv"
-    local long_opts="help,update-patches-only,no-bugs,skip-branch-name-check,skip-release,sru-bug,verbose"
+    local long_opts="help,update-patches-only,no-bugs,first-sru,skip-branch-name-check,skip-release,sru-bug:,verbose"
```

Comment:
is the `:` supposed to be on `sru-bug:` ?

Reply:
@TheRealFalcon yes, that ":" to getopt tells getopt that sru-bug requires a value as the followin parameter on the command-line. It was an oversight and bug in new-upstream-snapshot which was overlooked by performing the following:
```diff
-               --sru-bug) sru_bug_fillstr=$3; shift;;
+               --sru-bug) sru_bug_fillstr=$next; shift;;
```

In the above, getopt wasn't processing the <sru-bug-id> --sru-bug <sru-bug-id> so our parsing was sourcing $3 instead of $next which is actually $2.   Once we add the colon ':' to 'sru-bug:' in the long_opts it tells getopt to expect a required positional argument after --sru-bug 

Per the `man getopt`
```bash
       -l, --longoptions longopts
              The long (multi-character) options to be recognized.  More than one option name may be specified at once, by separating the names  with  commas.   This
              option  may  be given more than once, the longopts are cumulative.  Each long option name in longopts may be followed by one colon to indicate it has a
              required argument, and by two colons to indicate it has an optional argument.
```